src/daemon/daemon.cpp

```diff
@@ -373,14 +386,24 @@ std::vector<mp::NetworkInterface> validate_extra_interfaces(const mp::LaunchRequ
 
     if (!request->network_options().empty())
     {
+        std::string full_name =
+            (request->remote_name().empty() ? "release" : request->remote_name()) + ':' + request->image();
+
+        if (no_bridging_images.find(full_name) != no_bridging_images.end() &&
+            std::find_if(request->network_options().begin(), request->network_options().end(), [](const auto& net) {
+                return net.mode() == multipass::LaunchRequest_NetworkOptions_Mode_AUTO;
+            }) != request->network_options().end())
+        {
+            throw std::runtime_error(fmt::format("Automatic bridge configuration not implemented for {}", full_name));
+        }
+
```

Comment:
I understand you're doing this here to avoid wasting wasting cycles in network listing. However, if I try this in a backend that does not support bridging, I now get:
```
$ multipass launch --network en0 xenial
launch failed: Automatic bridge configuration not implemented for release:xenial
```

I wonder if it's worth wasting one or two seconds and check the mode only in the loop below, along with other validations. That would also make the code cleaner. You could drop the initial `if` entirely and get the networks in the first iteration of the loop, like so (pseudocode):

```C++
auto host_networks = [&factory](){
  static auto networks = factory.networks(); // called only once
  return networks;
};

for (...)
{
  auto networks = host_networks();
  ...
} 
```

WDYT? Worth it? Separate PR?

Reply:
Yeah, I think it's worth it. But I'll separate the code in a new commit.

Reply:
Actually, scratch that approach using static, my mistake. I was thinking that would be local to the lambda object, but no, it's local to the lambda's `operator()`, so the static will stay for the remainder of the program. (Even if such caching was desired, it should be done centrally, not here). 

Perhaps the simplest here is just an optional (to distinguish from the case where no networks are found). Something like:

```C++
optional<decltype(factory.networks())> networks = nullopt;
for (...)
{
  if (!networks)
    networks = factory.networks();
  ...
}
```

Reply:
About caching, I don't think it's a good idea. There is always the possibility of hotplugging devices, what will make caching useless. Unless we implement another mechanism to trigger an update on such events. In any case, out of the scope of this PR.

I was in doubt whether the scope of the `static` would be this function or global, too. Let me try the new approach, without the static thing.

Reply:
Done. The code looks better, indeed.

Reply:
> About caching, I don't think it's a good idea. There is always the possibility of hotplugging devices, what will make caching useless. Unless we implement another mechanism to trigger an update on such events. In any case, out of the scope of this PR.

Sure, I was not proposing, actually excluding it here :) I meant that this wouldn't be the way or place to achieve it, *even* if we wanted it. It's something we can discuss in the future, but disregard now.

> I was in doubt whether the scope of the static would be this function or global, too.

The scope is the function, but it still has static duration. Meaning it will be initialized in the first function call (`operator()`) and stay there, no matter what different objects (lambda instances) are used to call it. 

I also considered `[networks = factory.networks()] { return networks; }`, but then it would be called when defining the lambda, so outside the loop, which is what we want to avoid.